pages/index.js

```diff
@@ -275,16 +282,26 @@ export default function Home({
           <div className="mt-16 mb-8 space-y-16">
             {featuresDetails.map((feature, featureIdx) => (
               <div
+                ref={featureRefs[featureIdx][0]}
                 key={feature.name}
-                className="flex flex-col-reverse lg:grid lg:grid-cols-12 lg:items-center lg:gap-x-8"
+                className="flex flex-col-reverse overflow-x-hidden lg:grid lg:grid-cols-12 lg:items-center lg:gap-x-8"
               >
                 <div
-                  className={classNames(
+                  className={classNames(`
+                  ${
                     featureIdx % 2 === 0
-                      ? "lg:col-start-1"
-                      : "lg:col-start-8 xl:col-start-9",
-                    "mt-6 lg:mt-0 lg:row-start-1 lg:col-span-5 xl:col-span-4",
-                  )}
+                      ? `lg:col-start-1 ${
+                          featureRefs[featureIdx][1]
+                            ? "animate-fade-right"
+                            : ""
+                        }`
+                      : `lg:col-start-8 xl:col-start-9 ${
+                          featureRefs[featureIdx][1]
+                            ? "animate-fade-left"
+                            : ""
+                        }`
+                  }
+                  mt-6 lg:mt-0 opacity-0 lg:row-start-1 lg:col-span-5 xl:col-span-4`)}
```

Comment:
This is very difficult to read, using `classNames` makes it easier as the backtics are not required and nested ternaries add to the complexity.

Here is an example 
https://github.com/EddieHubCommunity/BioDrop/blob/090b58a1e91b461c7aa17ae839850fa2b3d968a6/components/Card.js#L8-L12

Reply:
Hey @eddiejaoude  I got your point but i don't understand that classNames so could you help me that how can i write that thing with classNames so that i can make those changes?

Reply:
> Hey @eddiejaoude I got your point but i don't understand that classNames so could you help me that how can i write that thing with classNames so that i can make those changes?

Okk Got it. Done with the changes